### PR TITLE
Replace zerograd(s) with cleargrad(s)

### PR DESCRIPTION
**Pull request by gwtnb**
`Variable.zerograd` requires allocated memory and fills with zero, but [these operations are not necessary](https://github.com/pfnet/chainer/blob/v1.13.0/chainer/variable.py#L375).

TODOs:
- ~~write tests~~
- write docs (I will send another PR for docs)

### CODE REVIEW

File: chainer/optimizer.py [hunk 1]
```diff
@@ -384,8 +384,12 @@ def update(self, lossfun=None, *args, **kwds):
 
         """
         if lossfun is not None:
+            use_cleargrads = kwds.pop('use_cleargrads', False)
```
Comment (beam2d) [inline chainer/optimizer.py]: I think it has no problem, but strictly speaking it breaks the compatibility. How do you think about adding a flag to the optimizer object itself?

File: chainer/link.py [hunk 2]
```diff
@@ -357,6 +357,16 @@ def copyparams(self, link):
         for name in self._params:
             dst[name].copydata(src[name])
 
+    def cleargrads(self):
+        """Clears all gradient arrays.
+
+        This method should be called before the backward computation at every
+        iteration of the optimization.
```
Comment (beam2d) [inline chainer/link.py]: The note is almost same as that of zerograds, so users may be confused. Would you update the docstring of zerograds? I think new users should use cleargrads in most cases.

**Comment by beam2d** [comment]
LGTM except above comments

File: chainer/optimizer.py (hunk 1 above)
Comment (gwtnb) [inline chainer/optimizer.py]: If the name `use_cleargrads` is already used, this PR breaks backward compatibility. I think it is too pessimistic to consider such case.

Comment (beam2d) [inline chainer/optimizer.py]: OK, I agree.

**Comment by beam2d** [comment]
LGTM

**Pull request merged**